pootle/apps/pootle_store/updater.py

```diff
@@ -123,7 +123,8 @@ def should_create_suggestion(self):
     def should_update_index(self):
         return (
             self.update.change_indices
-            and self.uid in self.update.indices)
+            and self.uid in self.update.indices
+            and self.unit.index != self.update.get_index(self.uid))
```

Comment:
with this update - the index would still get updated if required (not sure how gaps are related, but it would do the same as now in that respect.